dashboard/src/utils/table.ts

```diff
@@ -6,11 +6,9 @@ export const buildHardwareArray = (
   environment_compatible?: TestHistory['environment_compatible'],
   misc?: TestHistory['misc'],
 ): string[] | undefined => {
-  return environment_compatible
-    ? environment_compatible
-    : misc?.platform
-      ? [misc?.platform]
-      : undefined;
+  const miscArray: (string | undefined)[] = [misc?.platform];
+  const envArray = environment_compatible ?? [environment_compatible];
+  return miscArray?.concat(envArray).filter(i => i !== undefined && i !== null);
```

Comment:
I personally would do something like
```suggestion
  const miscArray: (string | undefined)[] = misc?.platform ?? [];
  const envArray = environment_compatible ?? [];
  return miscArray.concat(envArray);
```
now you don't need the filter